### ISSUE TITLE
Provide process type as $DYNO to container

### ISSUE THREAD

**Issue by yabawock** [progrium/dokku#733]
Heroku has a $DYNO environment variable in the container which shows which process from the Procfile is running in the container. It would be nice to have this variable available (even if hardcoded as „web“ like the start command) on Dokku too.

**Comment by josegonzalez** [comment]
Seems like we could set this by default - check if set and set it in pre-deploy - and use it [here](https://github.com/progrium/dokku/blob/master/dokku#L71). Thoughts, @michaelshobbs ? I don't know if we have a `dokku config:set --norebuild` command yet, but that's the only blocker I can think of.

**Comment by michaelshobbs** [comment]
To clarify, we'll modify `$START_CMD` if `$DYNO` is set in config? 

Do we want the default `config:set` action to rebuild or should that be explicit? I'm leaning towards `dokky config:set -r <app> <config>` to indicate rebuild and the default to not. Thoughts?

**Comment by josegonzalez** [comment]
I think we should always have it set and default it to `web` if it's not set.

Does heroku rebuild or not? If not, we should not, if so, we should, but we should allow people to do either/or.

**Comment by michaelshobbs** [comment]
According to https://devcenter.heroku.com/articles/config-vars a restart happens on set and unset.

**Comment by josegonzalez** [comment]
So we should at least do a restart (with the env vars sourced in). Is that not what we do now?

**Comment by michaelshobbs** [comment]
That is what we do now. Should we offer a flag to bypass the restart? If so, why?

**Comment by josegonzalez** [comment]
I don't know what I was thinking. Ignore that.

**Comment by joshco** [comment]
I'm bumping into this (or related) as well.  I use a Procfile in my app to specify multiple worker types.that all need to be running to service the app.

This is a rails app that has a web worker and a jobs worker (eg delayed_job)

The procfile is in the tree:
````
web: bundle exec  rails server -p $PORT
jobs: bundle exec rake jobs:work
````

For the moment, what I'm doing is manually starting a second docker for it:

> sudo docker run -d dokku/vwf /bin/bash -c "/start jobs"

But this doesn't seem optimal.  It seems like dokku should examine the procfile and do something useful.

I don't see the Heroku parallel metaphor here, which is their ps settings on each dyno/worker type.  In Heroku you specify the number of each dyno, and type/size you want to run.  The numbers arent part of the procfile, they are set via the heroku command.

> heroku ps:scale web=1, worker=2

That is what tells heroku what to do.

Is there a dokku idiom for this?

**Comment by michaelshobbs** [comment]
I use the dokku-logging-supervisor plugin to scale my processes at the moment.

**Comment by joshco** [comment]
Thanks for the tip.  I just tried it and it works very well!

**Comment by josegonzalez** [comment]
@joshco better would be for us to do actual process scaling - which would include this sort of feature. Is this something you're interested in working on?

**Comment by michaelshobbs** [comment]
So, I've been banging away on this one for a little bit. I had put it down as I ran into some implementation complexities that I haven't flesh out yet. I opened  a new issue (#1117) for discussion. @joshco (and anyone else for that matter) any additional input would be greatly appreciated.

**Comment by josegonzalez** [comment]
Closed by #1118
